Add blog post on using Miri in CI (#8)
kflansburg/kflansburg.github.io: content/posts/improving-rust-codebase-quality-with-miri-in-ci.md

---

> title: "Improving Rust Codebase Quality with Miri in CI"

Author: ```suggestion
title: "Detecting Undefined Behavior in Rust with Miri in GitHub Actions"
```

> date: 2023-10-02

Author: ```suggestion
date: 2024-06-01
```

draft: false
description: "Learn how to use Miri in CI to detect various types of errors in your Rust codebase and how to set up GitHub Actions for automatic PR runs."
tags:
  - Rust
  - CI/CD
  - Miri
  - GitHub Actions
---

> # Improving Rust Codebase Quality with Miri in CI

Author: ```suggestion
# Detecting Undefined Behavior in Rust with Miri in GitHub Actions
```

## Introduction to Miri

Miri is an interpreter for Rust's mid-level intermediate representation (MIR), which allows for the detection of undefined behavior and other errors at compile time. Integrating Miri into Continuous Integration (CI) workflows can significantly improve the quality of a Rust codebase by catching errors early in the development process. This can be especially important if your codebase requires the use of unsafe blocks of Rust code which may prevent the Rust compiler from catching bugs that you would normally expect it to.

> 

Author: ```suggestion

<!--more-->

```

## Types of Errors Detected by Miri

> You can find an extensive list of the kinds of errors that Miri can catch in [miri's test suite](https://github.com/rust-lang/miri/tree/master/tests).

Author: ```suggestion
You can find an extensive list of the kinds of errors that Miri can catch in [miri's test suite](https://github.com/rust-lang/miri/tree/master/tests). Note that many of these examples do not compile, and are used only to explain the bug.
```

---
title: "Detecting Undefined Behavior in Rust with Miri in GitHub Actions"
date: 2024-06-01
draft: false
description: "Learn how to use Miri in CI to detect various types of errors in your Rust codebase and how to set up GitHub Actions for automatic PR runs."
tags:
  - Rust
  - CI/CD
  - Miri
  - GitHub Actions
---

# Detecting Undefined Behavior in Rust with Miri in GitHub Actions

## Introduction to Miri

Miri is an interpreter for Rust's mid-level intermediate representation (MIR), which allows for the detection of undefined behavior and other errors at compile time. Integrating Miri into Continuous Integration (CI) workflows can significantly improve the quality of a Rust codebase by catching errors early in the development process. This can be especially important if your codebase requires the use of unsafe blocks of Rust code which may prevent the Rust compiler from catching bugs that you would normally expect it to.

<!--more-->

> 

Author: ```suggestion

### Warning

This post was drafted by [GitHub Copilot Workspaces](https://githubnext.com/projects/copilot-workspace) to experiment with its capabilities.   I have reviewed it closely, but am not a Miri expert. In particular, the code examples for various bug types involve subtle use of Rust syntax and may not be entirely accurate.

```
